### PR TITLE
Release 2.6.0

### PR DESCRIPTION
**Pull request by benjamn**
Similar to [Release 2.5.0](https://github.com/apollographql/apollo-client/pull/4361), this PR tracks changes that are slated to be released in the next minor version of `apollo-client`.

PRs that introduce minor breaking changes should be merged into the `release-2.6.0` branch instead of `master`, so that we can keep releasing patch updates in the meantime. That doesn't mean that everything here is a minor breaking change by any means—just that this branch is slightly more experimental/unstable than `master`.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -17,7 +17,7 @@
     "coverage": "jest --verbose --coverage",
     "coverage:upload": "codecov",
     "danger": "danger run --verbose --dangerfile=./config/dangerfile.ts",
-    "deploy": "lerna publish -m \"chore: Publish\" && cd packages/apollo-client && npm run deploy",
+    "deploy": "lerna publish -m \"chore: Publish\" --dist-tag beta && cd packages/apollo-client && npm run deploy",
```
Comment (benjamn) [inline package.json]: ⚠️ Just wanted to flag that this commit should be reverted before `release-2.6.0` is merged into `master`.

**Comment by OurMajesty** [comment]
beta 5:  
when variables are changed, loading is false and networkStatus is 7, but the data is empty, variables are empty too
изображение
then it set status to 4 and loading to true  

and, for some reason, when state of parent component changes, it forces update of child with same "empty" result

**Comment by benjamn** [comment]
@OurMajesty You can probably guess what I'm going to say, but a reproduction or failing test case would be helpful here! Also, please feel free to open an issue, and we will put it in the [Release 2.6.0 milestone](https://github.com/apollographql/apollo-client/milestone/16).

**Comment by benjamn** [comment]
Just published the first release candidate of all Apollo Client packages for this release:
 - apollo-boost@0.4.0-rc.1
 - apollo-cache-inmemory@1.6.0-rc.1
 - apollo-cache@1.3.0-rc.1
 - apollo-client@2.6.0-rc.1
 - apollo-utilities@1.3.0-rc.1
 - graphql-anywhere@4.2.2-rc.1

You can also refer to these packages using the `@beta` version constraint.

**Comment by OurMajesty** [comment]
Sorry, but how this could be release candidate if there are still at least three bugs not fixed? (#4530, #4636, #4691)

**Comment by benjamn** [comment]
@OurMajesty We've fixed many bugs in this release, and no one will benefit from those fixes until we actually release it. After we release 2.6.0, we will keep fixing bugs in future patch/minor releases. Waiting to release software until all the bugs are fixed is a recipe for never releasing software.

Two of the three bugs you mentioned are at least partially fixed by this release, according to the reproductions that were provided, though I see that you are still experiencing some similar symptoms. I think it's important to ship the improvements sooner rather than later, unless the problems were introduced between 2.5.x and 2.6.0, or their fixes require breaking changes—in which case we should absolutely prioritize fixing them during this release candidate period.

**Comment by OurMajesty** [comment]
@benjamn Thank you for the reply, appreciate your contribution and sense of purpose

**Comment by benjamn** [comment]
@OurMajesty Thanks for flagging the remaining issues that you want to see fixed in this release!

Speaking of which… now would be a great time for others to chime in if there's anything broken in 2.5 or 2.6 that seems to have slipped through the cracks.

**Pull request merged**